specs-go/v1/config.go

```diff
@@ -75,6 +75,7 @@ type History struct {
 }
 
 // Image is the JSON structure which describes some basic information about the image.
+// This provides the `application/vnd.oci.image.config.v1+json` mediatype when marshalled to JSON.
```

Comment:
> `cat blob | some-tool --add-option` won't work properly because you cannot deal with multiple versions and also error handling

But:

```
$ DESCRIPTOR="$(oci-refs get some-image v1.0)"
$ DIGEST="$(echo "${DESCRIPTOR}" | jq .digest)"
$ MEDIA_TYPE="$(echo "${DESCRIPTOR}" | jq .mediaType)"
$ oci-cas get some-image "${DIGEST}" | some-tool --add-option --media-type "${MEDIA_TYPE}"
```

works fine.  And the only case I can think of where you *aren't* getting to the digest by walking down from a ref is still validating a blob before you push it into CAS.  And there you presumably know the type, and just want to be saved from typing it out on the command line (and as I've said, I'm ok keeping a self-describing `mediaType` if/where we want to support this).

> And since oci-refs and oci-cas appear to be on the path to "images are just a CAS" you need to have that.

I think “images can be stored in CAS as a Merkle tree” is true, but I don't think they're *just* a set of Merkle blobs.  Otherwise we could have written `descriptor.md` and `image-layout.md` and skipped `manifest.md`, `manifest-list.md`, `layer.md`, etc.  CAS (and refs) are one level in the implementation, but there are additional, higher levels with more domain-specific spec.  I'd rather have image-tools provide APIs and tools for handling images at all of these levels.